### PR TITLE
New character scenario reset HP bugfix

### LINKED ISSUE #27460: Character creation HP bug
**Describe the bug**  
The default HP value for 8 strength is 84, however, when you start character creation, and pick challenge-abandoned scenario and then pick anything else, your HP will be halved, as if frail trait was assigned to you.
P.S. Same bugs probably could apply to any other scenario with default traits at the start (e.g. prison break with psychopath)

**To Reproduce**  
1. Pick challenge-abandoned scenario
2. Pick something different
3. Skip to "Stats" tab, to check your HP

**Expected behavior**  
HP value must be reset to default number, when changing to scenario without "frail" trait at the start.

**Versions and configuration(please complete the following information):**
 - OS: Windows 10
 - Game Version: 0.C-36161-g4f267bf
 - Graphics version: Tiles
 - No mods, DDA only

### PR DESCRIPTION
**Pull request by ghost**
#### Summary
```SUMMARY: bugfixes "Fix HP not being recalculated during chargen'"```

#### Purpose of change
Fixes #27460 

#### Describe the solution
Probably not ideal, an IF-check for HP-affecting traits , and running toggle_trait() on them.
it works, but it's cobbled together from the little I know.

#### Describe alternatives you've considered
In issue #27460 I detailed how I tried other things that didn't work.
Ideal would be to toggle all currently selected traits, as apparently 

```
for( auto &mut : my_mutations ) {
        mutation_loss_effect( mut.first );
```

is not enough.
but I don't know how to do that.

#### Additional context
N/A

### CODE REVIEW

**Comment by AMurkin** [comment]
You can probably do this:
```cpp
    for( auto &t : u.get_mutations() ) {
        if( t.obj().hp_modifier != 0 ) {
            u.toggle_trait( t );
        }
    }
```

**Comment by davidpwbrown** [comment]
> You can probably do this:
> 
> ```c++
>     for( auto &t : u.get_mutations() ) {
>         if( t.obj().hp_modifier != 0 ) {
>             u.toggle_trait( t );
>         }
>     }
> ```

That is better, incase hp-modifying traits are added in future.

File: src/newcharacter.cpp [hunk 1]
```diff
@@ -2620,7 +2620,18 @@ void reset_scenario( player &u, const scenario *scen )
     u.per_max = 8;
     g->scen = scen;
     u.prof = &default_prof.obj();
+    /// check for HP-modifying traits and toggle them
+    if( u.has_trait( trait_id( "FLIMSY" ) ) ) {
+      u.toggle_trait( trait_id( "FLIMSY" ) );
```
[the remark this answers is not in the document]
Comment (davidpwbrown) [reply src/newcharacter.cpp]: Just tested the alternative in the comment below as a better solution

Which works, so commiting that instead

**Comment by Night-Pryanik** [comment]
As far as I can tell, you shouldn't remove `on_mutation_loss`, because it is used only for lua.

**Comment by davidpwbrown** [comment]
> As far as I can tell, you shouldn't remove `on_mutation_loss`, because it is used only for lua.

Ahh I wondered why that was there, it didn't seem to do anything, I thought it was a mistaken attempt to remove the trait properly, so I replaced it, but I'll put it back.

**Pull request merged**